src/runner.ts

```diff
@@ -0,0 +1,157 @@
+import * as core from "@actions/core";
+import * as types from './types';
+
+import { messages } from './messages';
+
+export interface RunDetails {
+    exitCode: number;
+}
+
+export class CoverageParserRunner {
+    async run() : Promise<RunDetails> {
+        // TODO: Simulate coverageNode input for testing the structure implemented in current task
+        const coverageNode = this.getCoverageNode();
+
+        await this.generateCoverageSummary(coverageNode);
+        return { exitCode: 0 };
+    }
+
+    private async generateCoverageSummary(coverageNode: types.CoberturaCoverage | null) {
+        if (!coverageNode) {
+            throw new Error(messages.invalid_coverage_data);
```

Comment:
Please discuss with @rwang-parasoft whether we should provide a more detailed error message indicating what is invalid. As we discussed yesterday, the Cobertura report is an intermediate file. However, even so, we need to provide useful information to the user when an error occurs. The current error message feels a bit too general.

Reply:
A related question is whether we should upload the intermediate Cobertura report as an artifact in this case, given that the error could occur during the conversion.

Reply:
I think we should not throw error. Sine we will handle it when getting data from cobertura report, it should never be null.